operators/multiclusterobservability/controllers/placementrule/manifestwork.go

```diff
@@ -423,10 +457,135 @@ func createManifestWorks(
 
 	work.Spec.Workload.Manifests = manifests
 
-	err = createManifestwork(c, work)
+	if clusterName != clusterNamespace && os.Getenv("UNIT_TEST") != "true" {
+		// ACM 8509: Special case for hub/local cluster metrics collection
+		// install the endpoint operator into open-cluster-management-observability namespace for the hub cluster
+		log.Info("Creating resource for hub metrics collection", "cluster", clusterName)
+		err = createUpdateResourcesForHubMetricsCollection(c, manifests)
+	} else {
+		err = createManifestwork(c, work)
+	}
+
 	return err
 }
 
+func createCSR() ([]byte, []byte) {
```

Comment:
seems it's not a correct choice to use CSR flow here. Have you check the possibility to use the functions in https://github.com/stolostron/multicluster-observability-operator/blob/main/operators/multiclusterobservability/pkg/certificates/certificates.go?

Reply:
maybe just generate a private key and then sign it to have the client cert.